wnfs/src/private/file.rs

```diff
@@ -364,6 +427,60 @@ impl PrivateFile {
         })
     }
 
+    /// Drains the content streamed-in and puts it into the private forest
+    /// as blocks of encrypted data.
+    /// Returns an external `FileContent` that contains necessary information
+    /// to later retrieve the data.
+    pub(super) async fn prepare_content_streaming(
+        bare_name: &Namefilter,
+        mut content: impl AsyncRead + Unpin,
+        forest: &mut Rc<PrivateForest>,
+        store: &mut impl BlockStore,
+        rng: &mut impl RngCore,
+    ) -> Result<FileContent> {
+        let key = SnapshotKey(AesKey::new(get_random_bytes(rng)));
+
+        let mut block_index = 0;
+
+        loop {
+            let mut current_block = vec![0u8; MAX_BLOCK_SIZE];
+            let nonce = SnapshotKey::generate_nonce(rng);
+            current_block[..NONCE_SIZE].copy_from_slice(&nonce);
+
+            // read up to MAX_BLOCK_CONTENT_SIZE content
+
+            let content_end = NONCE_SIZE + MAX_BLOCK_CONTENT_SIZE;
```

Comment:
Worth making into a constant?

Reply:
Hm, I don't think so. At least not a constant expose at the top level.